leader-election-impl/src/main/java/com/palantir/paxos/VerifyingPaxosStateLog.java

```diff
@@ -175,12 +201,15 @@ private long getExtremeLogEntry(Function<PaxosStateLog<V>, Long> extractor) {
         lock.readLock().lock();
         try {
             Long result = extractor.apply(currentLog);
-            Long experimentalResult = safeReadFromExperimental(extractor::apply);
-            if (!Objects.equals(result, experimentalResult) && result != PaxosAcceptor.NO_LOG_ENTRY) {
-                log.error("Mismatch in getting the extreme log entry between legacy and current implementations."
-                                + " Legacy result {}, current result {}.",
-                        SafeArg.of("legacy", result),
-                        SafeArg.of("current", experimentalResult));
+            if (firstEntry.isDone()) {
+                Long experimentalResult = safeReadFromExperimental(extractor::apply);
+                if (!Objects.equals(result, experimentalResult) && result != PaxosAcceptor.NO_LOG_ENTRY
+                        && result >= Futures.getUnchecked(firstEntry)) {
```

Comment:
We probably don't need the second term here, but this is fine

Reply:
In hindsight, following a suggestion from @j-baker, it might be worth not doing this check actually, since this would be a good way to test a hypothesis that the old entries aren't actually particularly important. It might still be worth keeping track of the first entry and dumping that as an arg in the result.